tajo-core/src/main/java/org/apache/tajo/master/exec/DDLExecutor.java

```diff
@@ -446,11 +447,35 @@ public void alterTable(TajoMaster.MasterContext context, final QueryContext quer
     case SET_PROPERTY:
       catalog.alterTable(CatalogUtil.setProperty(qualifiedName, alterTable.getProperties(), AlterTableType.SET_PROPERTY));
       break;
+    case ADD_PARTITION:
+      existColumnNames(qualifiedName, alterTable.getColumnNames());
+      catalog.alterTable(CatalogUtil.addPartitionAndDropPartition(qualifiedName, alterTable.getColumnNames(),
+        alterTable.getPartitionValues(), alterTable.getLocation(), AlterTableType.ADD_PARTITION));
+      break;
+    case DROP_PARTITION:
+      existColumnNames(qualifiedName, alterTable.getColumnNames());
+      catalog.alterTable(CatalogUtil.addPartitionAndDropPartition(qualifiedName, alterTable.getColumnNames(),
+        alterTable.getPartitionValues(), alterTable.getLocation(), AlterTableType.DROP_PARTITION));
+      break;
     default:
       //TODO
     }
   }
 
+  private boolean existColumnNames(String tableName, String[] columnNames) {
+    for(String columnName : columnNames) {
+      if (!existPartitionColumnName(tableName, columnName)) {
+        throw new NoSuchColumnException(columnName);
```

Comment:
This case seems to indicate that the given table has a given column, but not as a partition key. If so, ```NoSuchColumnException``` makes users confused.
How about make a new exception for it such as ```NoSuchPartitionKeyException```?